wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt

```diff
@@ -173,14 +155,41 @@ class MainViewModel @Inject constructor(
         }
     }
 
+    private fun updateEntityStates(entity: Entity<*>) {
+        if (supportedDomains().contains(entity.domain)) {
+            entities[entity.entityId] = entity
+            // add to cache if part of favorites
+            if (favoriteEntityIds.value.contains(entity.entityId)) {
+                addCachedFavorite(entity.entityId)
+            }
+        }
+    }
+
+    suspend fun updateUI() = withContext(Dispatchers.IO) {
+        val getAreaRegistry = async { homePresenter.getAreaRegistry() }
+        val getDeviceRegistry = async { homePresenter.getDeviceRegistry() }
+        val getEntityRegistry = async { homePresenter.getEntityRegistry() }
+        val getEntities = async { homePresenter.getEntities() }
+
+        areaRegistry = getAreaRegistry.await()?.also {
+            areas.addAll(it)
+        }
+        deviceRegistry = getDeviceRegistry.await()
+        entityRegistry = getEntityRegistry.await()
+
+        getEntities.await()?.forEach {
+            updateEntityStates(it)
```

Comment:
Any entities currently in `entities` but not in this response also need to be removed when the app is resumed